### PR TITLE
[hailctl/dataproc] Support Broad's GCP Security Best Practices Document

### PR DESCRIPTION
**Pull request by danking**
These changes enable hailctl clusters to work correctly in an Broad GCP Security Best Practices configured project, with minor hail-specific set-up. See further details in team chat.

### CODE REVIEW

File: hail/python/hailtop/hailctl/dataproc/start.py [hunk 1]
```diff
@@ -196,3 +200,8 @@ def disk_size(size):
     if not args.dry_run:
         print("Starting cluster '{}'...".format(args.name))
         sp.check_call(cmd)
+
+    if args.master_tags:
```
Comment (tpoterba) [inline hail/python/hailtop/hailctl/dataproc/start.py]: tab over to be inside the execution path (dont execute with dry-run)

**Comment by chrisvittal** [comment]
How does this prevent logging in to the workers?

**Comment by danking** [comment]
See the separate document in `team` about firewall-rules. In particular, the default network will block all connections.

**Comment by chrisvittal** [comment]
Understood. Can we tag every node rather than just the master?

**Comment by danking** [comment]
Dataproc does have an option to tag all nodes (`--tags`), but we purposely intend to prevent SSH connections from the outside world to the worker nodes. If you need to access the worker machines, you can still connect to them from the worker node (because, due to Spark/Dataproc, we must permit all TCP/IP connections between all Dataproc nodes). I realize this is annoying, but I have only needed to do this once in my three years of working on Hail.

**Comment by danking** [comment]
In particular the design of the Broad GCP Security Best Practices is to disable everything by default and then selectively enable. The master tag enables SSH access for the master node.

**Pull request merged**